configure.ac

```diff
@@ -4466,21 +4466,23 @@ case "$host" in
 			TARGET=POWERPC64;
 			CPPFLAGS="$CPPFLAGS -D__mono_ppc__ -D__mono_ppc64__"
 			# mono#18554 - be more robust in testing for -mminimal-toc
-			AC_MSG_NOTICE([Checking PowerPC ABI])
-			if ! (echo $CC | grep -q -- 'clang'); then
-				if ! (echo | cc -dM -E - | awk '/_CALL_ELF/ {print $NF}'); then
-					AX_CHECK_COMPILE_FLAG(
-						[-mminimal-toc],
-						[CFLAGS="$CFLAGS -mminimal-toc"],
-						[CFLAGS="$CFLAGS"]
-					)
-					AC_DEFINE([POWERPC_ELF], 1, [PowerPC ELFv1])
-				else
-					# Do not set -mminimal-toc on ELFv2 systems
-					AC_DEFINE([POWERPC_ELFV2], 1, [PowerPC ELFv2])
-					CFLAGS="$CFLAGS"
-				fi
-			fi
+			AC_MSG_NOTICE([Checking for PowerPC ISA -mminimal-toc support])
+			AX_CHECK_COMPILE_FLAG(
+				[-mminimal-toc],
+				[CFLAGS="$CFLAGS -mminimal-toc"],
+				[CFLAGS="$CFLAGS"]
+			)
+			case "$host" in
+				powerpc*-*-freebsd*)
```

Comment:
This is wrong.
1. ELFv1 and ELFv2 are only on powerpc64. powerpc (32-bits) has still only one ABI, so this check shouldn't be for powerpc*, but powerpc64.
2. You check for freebsd*, but ELFv1 and ELFv2 are not FreeBSD-ism. All powerpc64 systems use either ELFv1 or ELFv2 ABI. AIX is strictly ELFv1, FreeBSD is migrating to ELFv2 and most Linux distros are ELFv2 now.

So this check should be:
`powerpc64-*-*)`

Consequently, you shouldn't print message "Checking FreeBSD ELF version", but e.g. "Checking ELF ABI version".